elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -238,14 +238,15 @@ public void commit() {
             joinCriteria(sessionCriteria, loadClass);
         }
 
+        sessionCriteria.setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY);
         if (!isScrollEnabled || isJoinQuery()) {
             return sessionCriteria.list();
         }
         return new ScrollableIterator(sessionCriteria.scroll(scrollMode));
     }
 
     /**
-     * Should this transaction use JOINs. By default will be enabled if Sparse fields are available.
+     * Should this transaction use JOINs. By default will be enabled only if parameter <code>join=true</code>.
```

Comment:
I wonder if this should be a server parameter since it can be dangerous if not properly configured.